packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -402,24 +412,53 @@ function buildTree(rootStack, readHookLog): HooksTree {
       subHooks: [],
     });
   }
+
+  // Associate custom hook values (useInpect() hook entries) with the correct hooks
+  rootChildren.forEach(hooksNode => rollupDebugValues(hooksNode));
+
   return rootChildren;
 }
 
+function rollupDebugValues(hooksNode: HooksNode): void {
```

Comment:
What does this function do exactly? I struggle to understand just from reading the code — maybe because of `filter` with a mutation and recursion inside. Not saying it needs to be changed but a brief comment about what it does to the tree would help.